### ISSUE TITLE
Why does Mongoose choose not to save arrays when an index is modified?

### ISSUE THREAD

**Issue by anl176** [Automattic/mongoose#9371]
According to the first question on the [FAQ](https://mongoosejs.com/docs/faq.html#), Mongoose does not set getters and setters for array indexes. My team got around this by including a check for arrays on objects we want to save

```
let schemaFields = Object.keys(document.schema.obj);
        _(schemaFields).each(function(field){
            if(Array.isArray(document.get(field))){
                document.markModified(field);
            }
        });
       
document.save((error, result) => {
    if (error) return reject(error);
    
    resolve(result);
})
```

A very simple fix but the simplicity of it makes me worried about why the Mongoose team does not have a similar workaround if it's so simple. I tried searching but couldn't find anything about this design decision. Does Mongoose not do this because of performance reasons? Is there a landmine Mongoose avoided that I'm walking into?

**Comment by AbdelrahmanHafez** [comment]
If I understand the example you wrote correctly, you're marking all the fields as modified regardless of whether they were actually modified or not, which will send an update/insert to MongoDB even though it might have been unnecessary.

**Comment by vkarpov15** [comment]
@AbdelrahmanHafez is correct: Mongoose tries to send a minimal update to MongoDB when you call `save()`. Your code, as written, will always mark all arrays as modified, meaning you'll always send the whole array when you `save()`.

We're working on a fix for this issue in v6.0 using proxies, follow #9290 for updates.
